src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,291 @@
+                document,
+                selection,
+                node: declarationSyntax,
+                holes: ImmutableArray.Create(declarator.Initializer.Value.Syntax),
+                cancellationToken).ConfigureAwait(false);
+
+            if (!isValidSelection)
+            {
+                return default;
+            }
+
+            if (!IsLegalUsingStatementType(semanticModel.Compilation, localType))
+            {
+                return default;
+            }
+
+            return (declarationSyntax, declarator.Symbol);
+        }
+
+        private static bool IsLegalUsingStatementType(Compilation compilation, ITypeSymbol type)
+        {
+            var disposableType = compilation.GetSpecialType(SpecialType.System_IDisposable);
+            if (disposableType is null)
+            {
+                return default;
+            }
+
+            // CS1674: type used in a using statement must be implicitly convertible to 'System.IDisposable'
+            return compilation.ClassifyCommonConversion(type, disposableType).IsImplicit;
+        }
+
+        private async Task<Document> IntroduceUsingStatementAsync(
+            Document document,
+            TextSpan span,
+            CancellationToken cancellationToken)
+        {
+            var (declarationStatement, localVariable) = await FindDisposableLocalDeclaration(document, span, cancellationToken).ConfigureAwait(false);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (!(declarationStatement.Parent is TBlockSyntax parent))
+            {
+                throw new NotImplementedException();
+            }
+
+            var syntaxFactsService = document.GetLanguageService<ISyntaxFactsService>();
+
+            var lastUsageStatement = FindSiblingStatementContainingLastUsage(
+                declarationStatement,
+                localVariable,
+                semanticModel,
+                document.GetLanguageService<ISyntaxFactsService>(),
+                cancellationToken);
+
+            var parentStatementsList = GetStatements(parent);
+
+            var declarationStatementIndex = parentStatementsList.IndexOf(declarationStatement);
+
+            var lastUsageStatementIndex = lastUsageStatement is null
+                ? -1
+                : parentStatementsList.IndexOf(lastUsageStatement, declarationStatementIndex + 1);
+
+            var statementsToSurround = GetStatementsToSurround(parentStatementsList, declarationStatementIndex, lastUsageStatementIndex);
+
+
+            var trailingTrivia = SplitTrailingTrivia(declarationStatement, syntaxFactsService);
+
+            var usingStatement =
+                CreateUsingStatement(
+                    declarationStatement.WithoutTrivia(),
```

Comment:
i think your C# specific code also stripped trivia.

Reply:
@CyrusNajmabadi It strips the trivia on `declarationStatement.Declaration`, not `delcarationStatement` itself. Tests caught it.